presto-kafka/src/main/java/io/prestosql/plugin/kafka/encoder/raw/RawRowEncoder.java

```diff
@@ -0,0 +1,344 @@
+        INT(Integer.SIZE),
+        LONG(Long.SIZE),
+        FLOAT(Float.SIZE),
+        DOUBLE(Double.SIZE);
+
+        private final int size;
+
+        FieldType(int bitSize)
+        {
+            this.size = bitSize / 8;
+        }
+
+        public int getSize()
+        {
+            return size;
+        }
+    }
+
+    private static final Pattern MAPPING_PATTERN = Pattern.compile("(\\d+)(?::(\\d+))?");
+    private static final Set<Type> SUPPORTED_PRIMITIVE_TYPES = ImmutableSet.of(
+            BIGINT, INTEGER, SMALLINT, TINYINT, DOUBLE, REAL, BOOLEAN);
+
+    public static final String NAME = "raw";
+
+    private final List<ColumnMapping> columnMappings;
+    private final ByteBuffer buffer;
+
+    public RawRowEncoder(ConnectorSession session, List<EncoderColumnHandle> columnHandles)
+    {
+        super(session, columnHandles);
+
+        for (EncoderColumnHandle handle : this.columnHandles) {
+            checkArgument(handle.getFormatHint() == null, "Unexpected format hint '%s' defined for column '%s'", handle.getFormatHint(), handle.getName());
+            checkArgument(isSupportedType(handle.getType()), "Unsupported column type '%s' for column '%s'", handle.getType().getDisplayName(), handle.getName());
+        }
+
+        this.columnMappings = buildMappings(this.columnHandles);
+
+        for (ColumnMapping mapping : this.columnMappings) {
+            checkMappingMatchesTypeSize(mapping);
+        }
+
+        int position = 0;
+        for (ColumnMapping mapping : this.columnMappings) {
+            checkMappingOffsets(mapping, position);
+            position += mapping.getLength();
+        }
+
+        this.buffer = ByteBuffer.allocate(position);
+    }
+
+    // parse column mappings from column handles
+    private static List<ColumnMapping> buildMappings(List<EncoderColumnHandle> columnHandles)
+    {
+        return columnHandles.stream().map(ColumnMapping::new).collect(toImmutableList());
+    }
+
+    private static void checkMappingMatchesTypeSize(ColumnMapping mapping)
+    {
+        if (mapping.getLength() != mapping.getFieldType().getSize() && !isVarcharType(mapping.getType())) {
+            throw new IndexOutOfBoundsException(format(
+                    "Mapping length '%s' is not equal to expected length '%s' for column '%s'",
+                    mapping.getLength(),
+                    mapping.getFieldType().getSize(),
+                    mapping.getName()));
+        }
+    }
+
+    // check that column mappings don't overlap and that there are no gaps
+    private static void checkMappingOffsets(ColumnMapping mapping, int expectedPosition)
+    {
+        checkArgument(mapping.getStart() == expectedPosition, format(
+                "Start mapping '%s' for column '%s' does not equal expected mapping '%s'",
+                mapping.getStart(),
+                mapping.getName(),
+                expectedPosition));
+        checkArgument(mapping.getEnd() > mapping.getStart(), format(
```

Comment:
`checkMappingOffsets()` is validating (in part) that the encoder itself is constructed properly.  The constructor of `ColumnMapping` is responsible for making sure that those objects are properly constructed.